### PR TITLE
Button to save in hard drive

### PR DESCRIPTION
**Pull request by sibi0002**

### CODE REVIEW

File: src/actions/reportsActions.js [hunk 1]
```diff
@@ -139,6 +140,23 @@ export const unfavoriteReportACreator = reportId => {
...
+export const downloadReportZipFile = (downloadReportZipFileStatus, downloadReportZipFileErrors) => {
+  return { type: DOWNLOAD_REPORT_ZIP_FILE, downloadReportZipFileStatus, downloadReportZipFileErrors}
+};
+
+export const downloadReportZipFileACreator = (fileName) => dispatch => {
+  return new Promise((resolve, reject) => {
+    WebApi.reports.get.reportZip(fileName).then(response => {
+      dispatch(downloadReportZipFile(true, []));
+      window.open(response.replyBlock.request.responseURL);
+      resolve();
```
Comment (BartusZak) [inline src/actions/reportsActions.js]: What is a reason of that resolve? We already have got returned data from api by promise.

File: src/actions/reportsActions.js [hunk 2]
```diff
@@ -227,6 +245,36 @@ export const generateReportACreator = (
...
+      saveAsFavorite: saveAsFavorite,
+      availableUntil: availableUntil
+    };
+
+    dispatch(setIsStarted(true, "Generowanie raportu"));
+      WebApi.reports.post
+        .report(model, false, false)
+        .then(response => {
+          dispatch(setIsStarted(false, ""));
+          dispatch(generateReport(true, []));
```
Comment (BartusZak) [inline src/actions/reportsActions.js]: - async dispatching, are u sure there is no need to do it synchronically ?

File: src/components/reports/ReportsContainer.js [hunk 3]
```diff
@@ -215,18 +218,22 @@ class ReportsContainer extends Component {
...
-      generateReport(
-        addList,
-        choosenFolder,
-        pagesList,
-        history,
-        this.state.saveAsFavorite,
-        this.state.availableUntilToggle ? this.state.availableUntilDate : null
-      );
+
+      if(this.state.hardDrive){
```
Comment (BartusZak) [inline src/components/reports/ReportsContainer.js]: use destructors for better dev experience and cleaner code

File: src/components/reports/ReportsContainer.js [hunk 4]
```diff
@@ -215,18 +218,22 @@ class ReportsContainer extends Component {
...
+        generateReportToHardDrive(addList, pagesList, this.state.saveAsFavorite, this.state.availableUntilToggle ? this.state.availableUntilDate : null)
+        .then(response => {
```
Comment (BartusZak) [inline src/components/reports/ReportsContainer.js]: use destructors for better dev experience and cleaner code

File: src/components/reports/ReportsContainer.js [hunk 5]
```diff
@@ -215,18 +218,22 @@ class ReportsContainer extends Component {
...
-      );
+
+      if(this.state.hardDrive){
+        generateReportToHardDrive(addList, pagesList, this.state.saveAsFavorite, this.state.availableUntilToggle ? this.state.availableUntilDate : null)
+        .then(response => { 
+          downloadReportZipFileACreator(response.replyBlock.data.dtoObject.filename);
+        }).catch(() => console.log("Error"));
+      }
+      else{
+        generateReport(addList, choosenFolder, pagesList, history, this.state.saveAsFavorite, this.state.availableUntilToggle ? this.state.availableUntilDate : null);
```
Comment (BartusZak) [inline src/components/reports/ReportsContainer.js]: use destructors for better dev experience and cleaner code

File: src/components/reports/ReportsContainer.js [hunk 6]
```diff
@@ -433,6 +447,7 @@ class ReportsContainer extends Component {
             handleAvailableUntil={this.handleAvailableUntil}
             availableUntilStartDate={moment()}
             availableUntilDate={this.state.availableUntilDate}
+            hardDrive={this.state.hardDrive}
```
Comment (BartusZak) [inline src/components/reports/ReportsContainer.js]: use destructors for better dev experience and cleaner code

File: src/components/reports/ReportsContainer.js [hunk 7]
```diff
@@ -433,6 +447,7 @@ class ReportsContainer extends Component {
             handleAvailableUntil={this.handleAvailableUntil}
             availableUntilStartDate={moment()}
             availableUntilDate={this.state.availableUntilDate}
```
Comment (BartusZak) [inline src/components/reports/ReportsContainer.js]: use destructors for better dev experience and cleaner code

File: src/actions/reportsActions.js (hunk 2 above)
Comment (sibi0002) [reply src/actions/reportsActions.js]: https://stackoverflow.com/questions/43276291/is-store-dispatch-in-redux-synchronous-or-asynchronous

Comment (BartusZak) [reply src/actions/reportsActions.js]: dispatch as a method is synchronous but the whole method is handled async, so on:
we are running 2 dispatches together and we don't know when they will return a result.
(ale teraz to już nie wiem, pamiętam że jak robiłem tak to mi sie krzaczyło)

Comment (BartusZak) [reply src/actions/reportsActions.js]: image
image

File: src/actions/reportsActions.js (hunk 1 above)
Comment (sibi0002) [reply src/actions/reportsActions.js]: resolve method give signal of successfull end of promise. It doesn't necessarily have to return something

**Pull request merged**